### PR TITLE
Handle locked wallet

### PR DESCRIPTION
**Pull request by cstenglein**
TODOs:

- [x] prevent closing of locking dialog
- [x] ~~Create issue to enable closing of locking dialog & make unlock possible from dashboard~~
- [x] Better feedback
- [x] Tests

Images: 
image

image

image

image

### CODE REVIEW

File: src/components/Home/TransactionCard/TransactionCard.tsx [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
 
+  if (walletLocked) {
+    return (
+      <div className="p-5 h-full">
+        <article className="bd-card flex flex-col transition-colors min-h-144 md:min-h-0">
+          <article className="h-full flex justify-center items-center">
+            <div>
+              <ClosedLockIcon className="h-6 w-6" />
+            </div>
+            WALLET LOCKED
```
Comment (escapedcat) [inline src/components/Home/TransactionCard/TransactionCard.tsx:34]: Should this be translated?

File: src/components/Shared/UnlockModal/UnlockModal.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+              </button>
+            </form>
+          </article>
+          {passwordWrong && (
+            <div className="my-5 text-red-500">Wrong password</div>
+          )}
+        </>
+      )}
+    </ModalDialog>,
+    MODAL_ROOT
```
Comment (escapedcat) [inline src/components/Shared/UnlockModal/UnlockModal.tsx:85]: What's this?

File: src/components/Shared/UnlockModal/UnlockModal.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                type="submit"
+                className="bd-button p-3 my-5"
+                disabled={!isValid}
+              >
+                {t("wallet.unlock")}
+              </button>
+            </form>
+          </article>
+          {passwordWrong && (
+            <div className="my-5 text-red-500">Wrong password</div>
```
Comment (escapedcat) [inline src/components/Shared/UnlockModal/UnlockModal.tsx:80]: Could be `p` I guess

File: src/pages/Home/Home.tsx [hunk 4]
```diff
@@ -29,38 +30,45 @@ export const Home: FC = () => {
   const [detailTx, setDetailTx] = useState<Transaction | null>(null);
   const [transactions, setTransactions] = useState<Transaction[]>([]);
 
-  const { logout } = appCtx;
+  const theme = darkMode ? "dark" : "light";
```
Comment (escapedcat) [inline src/pages/Home/Home.tsx:33]: Could be types, but not too important

File: src/pages/Home/Home.tsx [hunk 5]
```diff
@@ -29,38 +30,45 @@ export const Home: FC = () => {
...
-
-  const closeSendModalHandler = (confirmed?: boolean) => {
-    setShowSendModal(false);
-    if (confirmed) {
-      const theme = appCtx.darkMode ? "dark" : "light";
-      toast.success(t("tx.sent"), { theme });
+    if (!walletLocked) {
+      instance
+        .get("/lightning/list-all-tx?reversed=true")
+        .then((tx: any) => {
```
Comment (escapedcat) [inline src/pages/Home/Home.tsx:39]: This `any` is intentional I guess. Wanna add a comment for this?

**Comment by escapedcat** [comment]
Keine tests? 😬

File: src/components/Home/TransactionCard/TransactionCard.tsx (hunk 1 above)
Comment (cstenglein) [reply src/components/Home/TransactionCard/TransactionCard.tsx:34]: Whoops, thank you!

File: src/components/Shared/UnlockModal/UnlockModal.tsx (hunk 2 above)
Comment (cstenglein) [reply src/components/Shared/UnlockModal/UnlockModal.tsx:85]: Rendering Stuff in a different place at the dom.
https://reactjs.org/docs/portals.html

MODAL_ROOT leads to here: https://github.com/cstenglein/raspiblitz-web/blob/master/public/index.html#L31 so it's "outside" the main root

File: src/pages/Home/Home.tsx (hunk 5 above)
Comment (cstenglein) [reply src/pages/Home/Home.tsx:39]: Not intentional, thank you :+1: 
I usually use `any` at first to not break the flow, but sometimes I forget to add it :D

**Comment by cstenglein** [comment]
ok. ok will add tests :P

File: src/components/Shared/UnlockModal/UnlockModal.tsx (hunk 2 above)
Comment (escapedcat) [reply src/components/Shared/UnlockModal/UnlockModal.tsx:85]: Oh, didn't know. Would it beokto prefix this with `PORTAL_`?

File: src/pages/Home/Home.tsx (hunk 5 above)
Comment (escapedcat) [reply src/pages/Home/Home.tsx:39]: What we do in projects is to create an `any`-alias called `FixMe` which indicates, that this type needs to be defined later on. While `any` would mean it's intentional (and could/should be commented why).

**Comment by escapedcat** [comment]
Not sure how helpful a placeholder-text is which is the same as the `label` but not important.

**Review by escapedcat — APPROVED** [review]

**Pull request merged**